src/webhooks/twitchWebhookValidator.ts

```diff
@@ -0,0 +1,30 @@
+import { log, LogLevel } from '../common'
+import * as Crypto from 'crypto'
+
+export abstract class TwitchWebhookValidator {
+
+    public static validate(request, response, next) {
+
+        const givenSignature = request.headers['x-hub-signature'];
+      
+        if (!givenSignature) {
+            log(LogLevel.Error, `webhooks: validator - missing signature`)
+            return response.status(409).json({
+                error: 'Missing signature'
+            });
+        }
+
+        let digest = Crypto.createHmac('sha256',process.env.TWITCH_CLIENT_SECRET)
```

Comment:
We're using the Twitch client secret as the secret in the webhook. I'm wondering if that's a good (read: safe) idea.

The reason I think it's not is that when we register the webhook, we'll have to send the secret we want Twitch to use. I'm not feeling good about sending the Twitch client secret in that request since it could potentially be grabbed and used nefariously.

Should we generate a secret (maybe a GUID) to use? Send that GUID to Twitch and keep up with it on our end to use in validating these requests.

@parithon am I overthinking?

Reply:
Oh good point!  It should be a different secret.

Reply:
If it is using a GUID, would the GUID be generated via code or set in the .env file?  

I can amend this PR and remove the check on the /follow endpoint or adjust the code to do the GUID and updating the Twitch API class.  

As for the current live instance is not checking bit, it may be that currently used subscriptions are receiving secrets to validate even with no secret sent. (i.e. using empty string as the secret).   We could also place a check to see if secret is set in the configuration and only check if there is one.  

Provide some direction and I'll amend the PR.

Reply:
When we subscribe with a new GUID each time the bot is started will it overwrite the previous registration? If it'll overwrite, then I vote to use the generated GUID each time (less work for the end-user). However, if it doesn't then it would need to be in the .env

Reply:
Anytime we register it will overwrite the existing matching subscription. So I suppose we'll generate a GUID somewhere and pass it to both TwitchAPI (to register the webhooks) and to the validator? Or just have it globally available?

Reply:
Since the validator is related to twitch and it needs the generated secret anyways, I could fold the validator into the integration TwitchAPI class instead.  This way the whole integration of Twitch API is in one spot.   It'd be adding a function in that class.

Reply:
That sounds good to me